src/actions/receive-profile.js

```diff
@@ -1012,8 +1009,13 @@ export function getProfilesFromRawUrl(
 
     switch (dataSource) {
       case 'from-addon':
-        await dispatch(retrieveProfileFromAddon(true));
-        break;
+        await dispatch(retrieveProfileFromAddon());
+        // This is not an actual error. It's notifying it's caller to not handle
+        // url processing and finalizing profile view anymore since it's happening
+        // inside retrieveProfileFromAddon sequentially already.
+        throw new Error(
+          `We don't need to handle url process for profiles from addon`
+        );
```

Comment:
Here, we are throwing an error, to make catch that error in caller. After calling this function, it's gonna go here and pass the url processing in the first if branch: 
https://github.com/firefox-devtools/profiler/blob/762d6dd6a414dc798108cbf51bdde91d3a41fe82/src/components/app/UrlManager.js#L102-L105

Reply:
Since this is not an actual error, I would prefer to change the function signature to return something like:

`ThunkAction<Promise<Profile | null>>`

Or

`ThunkAction<Promise<{|profile: Profile, isFromAddon: boolean |}>>`

The error seems a bit hard to maintain.